src/org/labkey/test/util/DeferredErrorCollector.java

```diff
@@ -156,17 +156,21 @@ public int errorsSinceMark()
      * Then resets the error mark.
      *
      * @param screenshotName A string to identify screenshots; Will be included in screenshot filenames.
+     * @return True if a screenshot was not taken (i.e. no error was generated), false if there was an error and screenshot was taken.
      * @see #takeScreenShot(String)
      */
-    public void screenShotIfNewError(@NotNull String screenshotName)
+    public boolean screenShotIfNewError(@NotNull String screenshotName)
     {
+        boolean noError = true;
         if (errorsSinceMark() > 0)
         {
             final String s = artifactCollector.dumpPageSnapshot(screenshotName);
             allErrors.get(allErrors.size() - 1).setScreenshotName(s);
             artifactCollector.reportTestMetadata(s);
+            noError = false;
         }
         setErrorMark();
+        return noError;
```

Comment:
The `setErrorMark` call is a no-op if there aren't any errors so the variable isn't really necessary here (not sure why I put the call outside the if).
I'd also flip the return value; if I saw `if (checker().screenShotIfNewError())`, I would expect it to return `true` if it took a screenshot.
```suggestion
    public boolean screenShotIfNewError(@NotNull String screenshotName)
    {
        if (errorsSinceMark() > 0)
        {
            final String s = artifactCollector.dumpPageSnapshot(screenshotName);
            allErrors.get(allErrors.size() - 1).setScreenshotName(s);
            artifactCollector.reportTestMetadata(s);
            setErrorMark();
            return true;
        }
        return false;
```

Reply:
Changed to return true if a screenshot is taken. Updated comment as well.